lib/addons/aws-batch-on-eks/index.ts

```diff
@@ -0,0 +1,38 @@
+import { ClusterAddOn, ClusterInfo } from "../../spi";
+import { Stack } from "aws-cdk-lib";
+import { CfnServiceLinkedRole, IRole, Role } from "aws-cdk-lib/aws-iam";
+import { Construct } from "constructs";
+
+const BATCH = 'aws-batch';
+
+export class AwsBatchAddOn implements ClusterAddOn {
+  deploy(clusterInfo: ClusterInfo): Promise<Construct> {
+    const cluster = clusterInfo.cluster;
+
+    // Create the service role used by AWS Batch on EKS 
+    const batchOnEksSlr = new CfnServiceLinkedRole(cluster.stack, 'BatchServiceRole', {
+      awsServiceName: 'batch.amazonaws.com',
+    });
+
+    //Init the service role as IRole because `addRoleMapping` method does not
+    //support the CfnServiceLinkedRole type
+    const batchEksServiceRole: IRole = Role.fromRoleArn(
```

Comment:
You are using Role.fromArn which assumes the role already exists with the name `AWSServiceRoleForBatch`. What creates that role? I don't see you specifying that name anywhere.

Reply:
When the Compute Environment is created, it creates the role with that exact name. However, compute environment is currently created inside the teams and not in the addon. I'll need to think of a way to address this.

Reply:
So when we create the service role (line 13-15), it will automatically generate the exact ARN: `arn:aws:iam::<acccountid>:role/aws-service-role/batch.amazonaws.com/AWSServiceRoleForBatch` (no suffix is allowed for service role names)

The reason we truncate between `role` and `AWSServiceRoleForBatch` is because there is an issue with aws-auth if we use the full role ARN. See note at the bottom of Step 1 [here](https://docs.aws.amazon.com/batch/latest/userguide/getting-started-eks.html#getting-started-eks-step-1) and github issue on this [here](https://github.com/kubernetes-sigs/aws-iam-authenticator/issues/268).